### PR TITLE
executors: add manifests for native executors

### PR DESCRIPTION
**Pull request by danieldides**
TODO: link ticket

### Description

This PR adds in the manifests for native k8s executors as an option alongside docker-in-docker executors. As such, it slightly reorganizes the `executors` directory.

The manifests are ported nearly directly from: https://github.com/sourcegraph/deploy-sourcegraph-k8s/pull/26 with only slight tweaks made to the `executors.ConfigMap.yaml` file. I will try and keep any new changes to that PR migrated to this repo as well.

> NOTE:

### Checklist

- [ ] [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) updated
- [ ] [K8s Upgrade notes updated](https://github.com/sourcegraph/sourcegraph/blob/main/doc/admin/updates/kubernetes.md)
- [ ] Sister [deploy-sourcegraph-k8s](https://github.com/sourcegraph/deploy-sourcegraph-docker) change:
- [ ] Sister [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker) change:
- [ ] All images have a valid tag and SHA256 sum
- [ ] I acknowledge that [deploy-sourcegraph-k8s](https://github.com/sourcegraph/deploy-sourcegraph-k8s) is now the preferred Kubernetes deployment repository

### Test plan
Tested in personal cluster.

### CODE REVIEW

**Review by Piszmog — APPROVED** [review]
Overall looks good to me

File: configure/executors/k8s/executor.ConfigMap.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  # Can be "batches" or "codeintel"
+  # Either set this or EXECUTOR_QUEUE_NAMES.
+  EXECUTOR_QUEUE_NAME: ""
+  # Used configure which queues Executors will process.
+  # Can be "batches" or "codeintel" or "batches,codeintel"
+  # Either set this or EXECUTOR_QUEUE_NAME.
+  EXECUTOR_QUEUE_NAMES: "batches,codeintel"
+  EXECUTOR_KUBERNETES_RESOURCE_REQUEST_MEMORY: "5Gi"
+#  KUBERNETES_RUN_AS_USER: ""
+#  KUBERNETES_RUN_AS_GROUP: ""
```
Comment (sanderginn) [inline configure/executors/k8s/executor.ConfigMap.yaml:29]: These will fail to parse during validation at startup time if they are provided as empty strings. Even though they are commented out, might be worth making them `-1` (which is the executor default)

**Review by sanderginn — APPROVED** [review]
lgtm

File: configure/executors/k8s/executor.ConfigMap.yaml (hunk 1 above)
Comment (danieldides) [reply configure/executors/k8s/executor.ConfigMap.yaml]: Good catch!

**Pull request merged**